src/main/java/org/commcare/xml/QueryPromptParser.java

```diff
@@ -55,15 +61,41 @@ public QueryPrompt parse() throws InvalidStructureException, IOException, XmlPul
         XPathExpression exclude = xpathPropertyValue(excludeValueString);
         XPathExpression required = xpathPropertyValue(parser.getAttributeValue(null, ATTR_REQUIRED));
 
+        QueryPromptValidation validation = null;
         while (nextTagInBlock(NAME_PROMPT)) {
-            if (NAME_DISPLAY.equals(parser.getName().toLowerCase())) {
+            if (NAME_DISPLAY.equalsIgnoreCase(parser.getName())) {
                 display = parseDisplayBlock();
-            } else if (NAME_ITEMSET.equals(parser.getName().toLowerCase())) {
+            } else if (NAME_ITEMSET.equalsIgnoreCase(parser.getName())) {
                 itemsetBinding = parseItemset();
+            } else if (NAME_VALIDATION.equalsIgnoreCase(parser.getName())) {
+                validation = parseValidationBlock(key);
             }
         }
         return new QueryPrompt(key, appearance, input, receive, hidden, display,
-                               itemsetBinding, defaultValue, allowBlankValue, exclude, required);
+                itemsetBinding, defaultValue, allowBlankValue, exclude, required, validation);
+    }
+
+    private QueryPromptValidation parseValidationBlock(String key)
+            throws InvalidStructureException, XmlPullParserException, IOException {
+        String testStr = parser.getAttributeValue(null, ATTR_VALIDATION_TEST);
+        if (testStr == null) {
+            throw new InvalidStructureException("No test condition defined in validation for key " + key);
```

Comment:
nit: wording suggestion
```suggestion
            throw new InvalidStructureException("No test condition defined in validation for prompt: " + key);
```